File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        dict = {"I": 1, "V": 5, "X":10, "L":50, "C":100, "D":500, "M":1000}
        prev = 0
        sum = 0
        for i, c in enumerate(reversed(s)):
            c_n = dict[c]
            if prev > c_n:
                sum -= c_n
            else:
                sum += c_n
            prev = c_n
        return sum

File: 0013-roman-to-integer/0013-roman-to-integer.java
import java.util.HashMap;

class Solution {
    public int romanToInt(String s) {
        HashMap<Character, Integer> map = new HashMap<>();
        map.put('I', 1);map.put('V', 5);map.put('X', 10);map.put('L', 50);map.put('C', 100);map.put('D', 500);map.put('M', 1000);

        int sum = map.get(s.charAt(s.length()-1));
        char prev_ch = s.charAt(s.length()-1);
        for(int i = s.length()-2; i >= 0; i--){
            if(map.get(s.charAt(i)) >= map.get(prev_ch)){
                sum += map.get(s.charAt(i));
            }else{
                sum -= map.get(s.charAt(i));
            }
            prev_ch = s.charAt(i);
        }
        return sum;
    }
}